src/mame/heathzenith/h8.cpp

```diff
@@ -70,411 +56,73 @@ class h8_state : public driver_device
-	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_KEYBOARD ) PORT_NAME("8 LOAD") PORT_CODE(KEYCODE_8) PORT_CHAR('8')
-	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_KEYBOARD ) PORT_NAME("9 DUMP") PORT_CODE(KEYCODE_9) PORT_CHAR('9')
-	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_KEYBOARD ) PORT_NAME("+") PORT_CODE(KEYCODE_UP) PORT_CHAR('^')
-	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_KEYBOARD ) PORT_NAME("-") PORT_CODE(KEYCODE_DOWN) PORT_CHAR('V')
-	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_KEYBOARD ) PORT_NAME("* CANCEL") PORT_CODE(KEYCODE_ESC) PORT_CHAR('Q')
-	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_KEYBOARD ) PORT_NAME("/ ALTER RST") PORT_CODE(KEYCODE_EQUALS) PORT_CHAR('=')
-	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_KEYBOARD ) PORT_NAME("# MEM RTM") PORT_CODE(KEYCODE_MINUS) PORT_CHAR('-')
-	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_KEYBOARD ) PORT_NAME(". REG") PORT_CODE(KEYCODE_R) PORT_CHAR('R')
 INPUT_PORTS_END
 
-INPUT_CHANGED_MEMBER(h8_state::button_0)
-{
-	if (newval)
-	{
-		u8 data = m_io_keyboard[1]->read() ^ 0xff;
-		if (BIT(data, 5))
-		{
-			m_maincpu->reset();
-		}
-		else if (BIT(data, 6))
-		{
-			m_intr_socket->set_irq_level(1, ASSERT_LINE);
-		}
-	}
-}
-
-void h8_state::machine_reset()
-{
-	m_pwr_led = 0;
-	m_irq_ctl = 1;
-	m_cassbit = 1;
-	m_cass_data[0] = 0;
-	m_cass_data[1] = 0;
-	m_uart->write_cts(0);
-	m_uart->write_dsr(0);
-	m_uart->write_rxd(0);
-	m_cass_data[3] = 0;
-	m_ff_b = 1;
-}
-
 void h8_state::machine_start()
 {
-	m_digits.resolve();
-	m_mon_led.resolve();
-	m_pwr_led.resolve();
-	m_ion_led.resolve();
-	m_run_led.resolve();
-
-	save_item(NAME(m_digit));
-	save_item(NAME(m_segment));
-	save_item(NAME(m_irq_ctl));
-	save_item(NAME(m_ff_b));
-	save_item(NAME(m_cass_data));
-	save_item(NAME(m_cassbit));
-	save_item(NAME(m_cassold));
-}
-
-void h8_state::h8_inte_callback(int state)
-{
-	// operate the ION LED
-	m_ion_led = !state;
-	m_irq_ctl &= 0x7f | ((state) ? 0 : 0x80);
-}
-
-void h8_state::h8_status_callback(u8 data)
-{
-/* This is rather messy, but basically there are 2 D flipflops, one drives the other,
-the data is /INTE while the clock is /M1. If the system is in Single Instruction mode,
-a int20 (output of 2nd flipflop) will occur after 4 M1 steps, to pause the running program.
-But, all of this can only occur if bit 4 of port F0 is low. */
-
-	bool state = (data & i8080_cpu_device::STATUS_M1) ? 0 : 1;
-	bool c,a = BIT(m_irq_ctl, 7);
-
-	if (BIT(m_irq_ctl, 1))
-	{
-		if (!state) // rising pulse to push data through flipflops
-		{
-			c = !m_ff_b; // from /Q of 2nd flipflop
-			m_ff_b = a; // from Q of 1st flipflop
-			if (c)
-			{
-				m_intr_socket->set_irq_level(2, ASSERT_LINE);
-			}
-		}
-	}
-	else
-	{ // flipflops are 'set'
-		c = 0;
-		m_ff_b = 1;
-	}
-
-
-	// operate the RUN LED
-	m_run_led = state;
-}
-
-void h8_state::h8_level3_irq_callback(int state)
-{
-	m_intr_socket->set_irq_level(3, state);
 }
 
-TIMER_DEVICE_CALLBACK_MEMBER(h8_state::kansas_w)
+void h8_state::device_config_complete()
 {
-	m_cass_data[3]++;
+	auto p1_lookup = m_p1.lookup()->get_card_device();
```

Comment:
Suggest adding a comment to explain this code represents connecting the p1 and p2 cards through a cable which is separate from the backplane. This is obvious to you but might help a future developer understand why it's done this way.